birdsong/views/activate.py

```diff
@@ -0,0 +1,46 @@
+
+from django.conf import settings
+from django.utils.encoding import force_str
+from django.utils.http import urlsafe_base64_decode
+from django.shortcuts import get_object_or_404, render
+from django.http import Http404
+
+from wagtail.models import Site
+
+from birdsong.conf import BIRDSONG_ACTIVATION_TEMPLATE
+from birdsong.models import Contact
+
+
+def activate(request, cidb64, token):
+    """Subscription activation endpoint
+
+    :param request: Request sent to this endpoint
+    :type request: class:`requests.models.Request`
+    :param cidb64: Base64 encoded contact ID (e.g. "YTEwMjA0MzctMGQwYS00ZjBjLWE4MzctMWM2YjgzMDQ3ZjM4")
+    :type cidb64: str
+    :param token: Subscription Activation Token (e.g. "bkwxpq-cc9a685f0e58c20baacf0ce2c93823f3")
+    :type token: str
+
+    :return: Rendered `activate.html` template as an HTTP Response (see `BIRDSONG_ACTIVATION_TEMPLATE`)
+    :rtype: class:`django.http.HttpResponse`
+    """
+    try:
+        cid = force_str(urlsafe_base64_decode(cidb64))
```

Comment:
uuids are url safe, was there a reason to base-64 encode it?

Reply:
Nice catch, the reason was to make it url safe but uuid typing will work just as well and perform better overall. Done.